src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinder.java

```diff
@@ -143,6 +153,24 @@ public SCMBinder(String scriptPath) {
         return new CpsScmFlowDefinition(scm, scriptPath).create(handle, listener, actions);
     }
 
+    private static Set<Class<? extends Cause>> passiveCauses = Set.of(
+        BranchIndexingCause.class,
+        BranchEventCause.class,
+        SCMTrigger.SCMTriggerCause.class,
+        TimerTrigger.TimerTriggerCause.class);
```

Comment:
given Jenkins extensibility, would it not be better to have a list of non passive causes to prevent any security issues in future?

Reply:
e.g. upstream / downstream cause?  Jira comment?
if using `withMaven` what if a new SNAPSHOT dependency was built elsewhere

Reply:
> a list of non passive (allowed) causes

Considered that, though it seemed more intrusive since for example `github-checks` (and the CloudBees CI equivalent) would both have to add to that list, which means either hard-coding class names or introducing a new extension point (in `scm-api`?) and having those plugins implement it. Something to consider.

> upstream / downstream cause?

Should not be triggering PR builds; or if it were, presumably for a reason (some other project called `build` on this PR branch project).

> Jira comment?

Not sure what you refer to.

> if using `withMaven` what if a new `SNAPSHOT` of a dependency was built elsewhere

Hmm, will need to check if that is in fact possible.

Reply:
> > Jira comment?
> Not sure what you refer to.

some jira plugin that allows to trigger builds based off a comment in Jira.